### PR TITLE
Implement DynamicQuantizeLinear

### LINKED ISSUE #380: Implement DynamicQuantizeLinear
Implement the onnx operator DynamicQuantizeLinear.
https://github.com/onnx/onnx/blob/master/docs/Operators.md#DynamicQuantizeLinear.

### PR DESCRIPTION
**Pull request by acrrd**
This PR implement the onnx operator [DynamicQuantizeLinear](https://github.com/onnx/onnx/blob/master/docs/Operators.md#DynamicQuantizeLinear).
I could not reuse `quantize_linear_u8` because it does `x * scale.recip()` instead of `x / scale`, This change some number enough to be rounded to another integer and the onnx testsuite fails.

Is not currently possible to implement it using the operations already provided (like is done in the `dynamicquantizelinear_expanded` test) because not all needed operations are `TypedOp` and in particular `QuantizeLinear` expect the zero point and the scale to be constant tensor.

Close #380

### CODE REVIEW

File: core/src/ops/quant.rs [hunk 1]
```diff
@@ -146,6 +147,110 @@ fn info_quantize_linear_i8(q: &QuantizeLinearI8) -> TractResult<Vec<String>> {
...
+        Ok(tvec!(quantized_tensor, scale_tensor, zero_point_tensor))
+    }
+}
+
+impl TypedOp for DynamicQuantizeLinearU8 {
+    fn output_facts(&self, inputs: &[&TypedFact]) -> TractResult<TVec<TypedFact>> {
+        let mut quantized_fact = inputs[0].clone();
+        quantized_fact.datum_type = u8::datum_type();
+        let shape = ShapeFact::from_dims(&[])?;
+        let scale_fact = TypedFact::dt_shape(f32::datum_type(), shape.clone())?;
```
Comment (kali) [inline core/src/ops/quant.rs:246]: I haven't tried it, but i think `TypedFact::dt_shape(f32::datum_type(), [].as_ref())?;` should work

**Review by kali — CHANGES_REQUESTED** [review]
Kudos on going so far on your own ! First time an outsider gives me a fully baked op without a huge amount of help :)

File: core/src/ops/quant.rs (hunk 1 above)
Comment (acrrd) [reply core/src/ops/quant.rs]: The compiler cannot infer the type with just that. I ended up with:
```
let shape: &[usize] = [].as_ref();
let scale_fact = TypedFact::dt_shape(f32::datum_type(), shape)?;
```

Comment (kali) [reply core/src/ops/quant.rs]: All right, thanks for trying 👍

**Review by kali — APPROVED** [review]

**Pull request merged**